libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/ManeuverMap.java

```diff
@@ -141,6 +141,21 @@
       R.drawable.ic_maneuver_roundabout_straight);
     maneuverMap.put(STEP_MANEUVER_TYPE_ROUNDABOUT, R.drawable.ic_maneuver_roundabout);
 
+    maneuverMap.put(STEP_MANEUVER_TYPE_ROUNDABOUT + STEP_MANEUVER_MODIFIER_LEFT + STEP_MANEUVER_MODIFIER_LEFT,
```

Comment:
what's the benefit of maintaining hashmap of the constants vs. simple `switch-case`? also note is that `ManeuverMap` constructor is not private, which may potentially result in multiple instances of the `ManeuverMap`object.

Reply:
> what's the benefit of maintaining hashmap of the constants vs. simple `switch-case`?

As Martin Fowler describes in _Refactoring Improving the Design of Existing Code_:

> The problem with switch statements is essentially that of duplication. Often you
find the same switch statement scattered about a program in different places. If you add a new clause to the switch, you have to find all these switch, statements and change them. The object oriented notion of polymorphism gives you an elegant way to deal with this problem.
>
> Most times you see a switch statement you should consider polymorphism.

So in general, switch statements are considered a bad smell 👀 https://refactoring.guru/smells/switch-statements

Most of the times a  switch statement can be easily replaced with map / dictionary.

And yeah, it's true there’s probably a bit more overhead in the dictionary than in the switch statement because you have to store the dictionary somewhere so it’s obviously using some memory. However IMO, in return you’re getting clean, concise, less error-prone and well-organized code that helps maintainability in the long term.